src/pages/playground/Playground.tsx

```diff
@@ -332,39 +354,28 @@ const Playground: React.FC<PlaygroundProps> = props => {
     [sessionId]
   );
 
-  const autorunButtons = React.useMemo(
-    () => (
+  const handleEditorEval = React.useCallback(
+    () => dispatch(evalEditor(workspaceLocation)),
+    [dispatch, workspaceLocation]
+  );
+
+  const autorunButtons = React.useMemo(() => {
+    return (
       <ControlBarAutorunButtons
-        handleDebuggerPause={props.handleDebuggerPause}
-        handleDebuggerReset={props.handleDebuggerReset}
-        handleDebuggerResume={props.handleDebuggerResume}
-        handleEditorEval={props.handleEditorEval}
-        handleInterruptEval={props.handleInterruptEval}
-        handleToggleEditorAutorun={props.handleToggleEditorAutorun}
-        isDebugging={props.isDebugging}
-        isEditorAutorun={props.isEditorAutorun}
-        isRunning={props.isRunning}
-        sourceChapter={props.playgroundSourceChapter}
+        {..._.pick(props, 'isDebugging', 'isEditorAutorun', 'isRunning', 'playgroundSourceChapter')}
```

Comment:
`playgroundSourceChapter` is an invalid prop for `ControlBarAutorunButtons`. This will result in a regression.

Reply:
Fixed in 1fe2721ad630909c60a3dbf7df2020cd09140ef6.